File: 4.二维数组中的查找/4.二维数组中的查找.py
# -*- coding:utf-8 -*-
class Solution:
    # array 二维列表
    def Find(self, target, array):
        # write code here
        if len(array[0])==0:
            return False
        row = len(array)
        col = len(array[0])
        xcol = 0
        xrow = row - 1
        while xrow > -1 and xcol < col:
            x = array[xrow][xcol]
            print x
            if x == target:
                return True
            elif x > target :
                xrow -= 1
            elif x < target :
                xcol += 1
        return False
            
            
            

File: 4.二维数组中的查找/4.二维数组中的查找.cpp
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
        if(array[0].size()==0){
            return false;
        }
        int row = array.size() - 1;
        int col = 0;
        while(row > -1 && col < array[0].size()){
            int x = array[row][col];
            if (x==target)
            	return true;
            else if (x > target)
            	row--;
            else
            	col++;
        }
        return false;
    }
};